cajita/unit_test/tstSparseIndexSpace.hpp

```diff
@@ -0,0 +1,681 @@
+                }
+    Kokkos::deep_copy( tile_insert_record, tile_insert_record_mirror );
+
+    Kokkos::View<bool[size][size][size], TEST_DEVICE> tile_label( "label" );
+
+    TEST_EXECSPACE().fence();
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, size ),
+        KOKKOS_LAMBDA( int tile_i ) {
+            for ( int tile_j = 0; tile_j < size; tile_j++ )
+                for ( int tile_k = 0; tile_k < size; tile_k++ )
+                {
+                    if ( Kokkos::atomic_fetch_or(
+                             &( tile_insert_record( tile_i, tile_j, tile_k,
+                                                    0 ) ),
+                             1 ) == 0 )
+                    {
+                        tile_label( tile_i, tile_j, tile_k ) = true;
+                    }
+                }
+        } );
+    TEST_EXECSPACE().fence();
+
+    auto tile_label_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), tile_label );
+
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+            {
+                EXPECT_EQ( tile_label_mirror( i, j, k ), true );
+            }
+}
+
+void testAtomicOrPro()
+{
+    constexpr int size = 64;
+
+    Kokkos::View<int*** [2], TEST_MEMSPACE> tile_insert_record(
+        "hash_record", size, size, size );
+
+    auto tile_insert_record_mirror =
+        Kokkos::create_mirror_view( Kokkos::HostSpace(), tile_insert_record );
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+                for ( int d = 0; d < 2; d++ )
+                {
+                    tile_insert_record_mirror( i, j, k, d ) = false;
+                }
+    Kokkos::deep_copy( tile_insert_record, tile_insert_record_mirror );
+
+    Kokkos::View<bool[size][size][size], TEST_DEVICE> tile_label( "label" );
+
+    TEST_EXECSPACE().fence();
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<TEST_EXECSPACE>( 0, size ),
+        KOKKOS_LAMBDA( int tile_i ) {
+            for ( int tile_j = 0; tile_j < size; tile_j++ )
+                for ( int tile_k = 0; tile_k < size; tile_k++ )
+                {
+                    if ( Kokkos::atomic_fetch_or(
+                             &( tile_insert_record( 0, 0, 0, 0 ) ), 1 ) == 0 )
+                    {
+                        tile_label( tile_i, tile_j, tile_k ) = true;
+                    }
+                }
+        } );
+    TEST_EXECSPACE().fence();
+
+    auto tile_label_mirror =
+        Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), tile_label );
+    int count = 0;
+    for ( int i = 0; i < size; i++ )
+        for ( int j = 0; j < size; j++ )
+            for ( int k = 0; k < size; k++ )
+            {
+                if ( tile_label_mirror( i, j, k ) == 1 )
+                    count++;
+            }
+
+    EXPECT_EQ( count, 1 );
+}
+
+void testTileSpace()
```

Comment:
Could this just be one test function templated on `size_bit` and `size`? To me the 2x2x2, 4x4x4, and 8x8x8 tests look the same except for the different `constexpr` integers.

Reply:
That works. I will change that